### PR TITLE
Outsource the prediction in a separate interface and class

### PR DESCRIPTION
**Pull request by jabenk**
- factor our the prediction

### CODE REVIEW

File: MREUnityRuntime/MREUnityRuntimeLib/Core/Physics/IPrediction.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MixedRealityExtension.Core;
+
+namespace MixedRealityExtension.Core.Physics
+{
+	/// interface struct to pass time step informations
+	public class PreditionTimeInterface
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/Core/Physics/IPrediction.cs:11]: maybe PredictionTimeParameters

Comment (jabenk) [reply MREUnityRuntime/MREUnityRuntimeLib/Core/Physics/IPrediction.cs]: Changed.

File: MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs [hunk 2]
```diff
@@ -208,8 +172,13 @@ public void FixedUpdate(UnityEngine.Transform rootTransform)
 					continue;
 				}
 
+				// <todo> make sure that this will always hit this loop ???
 				// Find corresponding rigid body info.
-				while (index < snapshot.RigidBodies.Count && rb.Id.CompareTo(snapshot.RigidBodies.Values[index].Id) > 0) index++;
+				index = 0;
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs]: RBs should be sorted for both arrays, we should not return index to 0 in every iteration

Comment (jabenk) [reply MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs]: yap.

**Pull request merged**